Prox/Prox/Data/TravelTimesProvider.swift

```diff
@@ -45,10 +49,33 @@ struct TravelTimesProvider {
             completion(travelTime)
         }
     }
+
+    static func travelTime(fromLocation: CLLocationCoordinate2D, toLocation: CLLocationCoordinate2D, byTransitTypes transitTypes: [MKDirectionsTransportType], withCompletion completion: @escaping ((TravelTimes?) -> ())) {
+        var allTimes = [TravelTimes?]()
+        for transitType in transitTypes {
```

Comment:
fwiw, I added the Deferred lib in [my branch](https://github.com/mcomella/prox/tree/server-comm-deferred/) but given this is a prototype, it's probably not worth changing something that works.

Note that the Deferred lib crashed with `.allFilled` (equivalent to Swift 2's `.all`) when returning optionals so it may not be usable here.